src/auro-input.js

```diff
@@ -286,15 +283,20 @@ export default class AuroInput extends BaseInput {
         aria-describedby="${this.uniqueId}"
         aria-invalid="${!this.isValid}"
       />
-
       <label for=${this.id} class="${classMap(labelClasses)}">${this.required ? this.label : `${this.label} (optional)`}</label>
       ${this.isValid
         ? html`
           <p class="inputElement-helpText" id="${this.uniqueId}">${this.helpText}</p>
         ` : html`
           <p class="inputElement-helpText error" id="${this.uniqueId}" role="alert" aria-live="assertive">${this.getErrorMessage()}</p>
         `}
-      ${this.type === 'credit-card' ? this.processCreditCard() : undefined}
+        ${this.type === 'credit-card' ? this.processCreditCard() : undefined}
+        ${this.type === 'credit-card' && this.icon ? html`<auro-icon class="creditCard-icon" category="payment" name="credit-card"></auro-icon>` : undefined}
+        ${this.inputIconName === 'cc-amex' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-amex"></auro-icon>` : undefined}
+        ${this.inputIconName === 'cc-visa' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-visa"></auro-icon>` : undefined}
+        ${this.inputIconName === 'cc-mastercard' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-mastercard"></auro-icon>` : undefined}
+        ${this.inputIconName === 'cc-discover' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-discover"></auro-icon>` : undefined}
+        ${this.inputIconName === 'cc-alaska' ? html`<auro-icon class="creditCard-icon" category="payment" name="cc-alaska"></auro-icon>` : undefined}
```

Comment:
So, I hacked on this a bit. While this works, it's a lot of repetition. To test, I created this function; 

```js
  ccIconType(ref) {
    if (ref === 'cc-visa') {
      return 'cc-visa'
    } else {
      return 'credit-card'
    }
  }
```

then then this update

```js
 ${this.type === 'credit-card' && this.icon ? html`<auro-icon class="creditCard-icon" category="payment" name="credit-card"></auro-icon>` : undefined}
 ${this.inputIconName ? html`<auro-icon class="creditCard-icon" category="payment" name="${this.ccIconType(this.inputIconName)}"></auro-icon>` : undefined}
```

When I run this code, I can see the `name` attribute change, but this is not triggering the change of the SVG output of the auro-icon CE. This feels like a problem we can solve. @geoffrich you have any ideas? 

![Screen Shot 2021-07-23 at 10 00 25 AM](https://user-images.githubusercontent.com/181089/126816554-5e4d03b7-adc5-430d-b2be-6a9b4b53fafe.png)

Reply:
This is what I ran into as well with my first solution attempt. I thought I could force it to re-render the icon using `requestUpdate()` but I wasn't able to get that to trigger the icon change. `name` is reflected in `auro-icon` as well.

If anyone has an idea on how to get the icon to render again after changing the name attribute I am all for using a single `auro-icon` block.

Reply:
Yeah, the trick that we are running into is that from auro-input we cannot get auro-icon to do another fetch request to get that icon. 

And to get this out the door, the scope doesn't include updating auro-icon, so we may be stuck. 

I have one last idea I want to try.

Reply:
Ok, so this works. There is some repetition that will happen again, but at least we can extract this logic from the template and put it into a function. 

```js
  ccIcon(ref) {
    let cardIcon = ''

    switch(ref) {
      case 'cc-visa':
        cardIcon = html`<auro-icon class="creditCard-icon" category="payment" name="${ref}"></auro-icon>`
        break;
      default:
        cardIcon = html`<auro-icon class="creditCard-icon" category="payment" name="credit-card"></auro-icon>`
    }

    return cardIcon;
  }
```

And then in the template, we can reduce it down to this one line 

```js
${this.inputIconName ? this.ccIcon(this.inputIconName) : undefined}
```

Reply:
Nope.... this is still stuck on the first render. E.g. when going from VISA to Alaska VISA, it doesn't switch. 

I am out of ideas. Fell free to disregard this feedback.

Reply:
I think updating auro-icon is the most straightforward approach here -- when `name` changes, re-fetch the icon. But if that's not possible, there's a hackier way using lit's [repeat directive](https://lit.dev/docs/templates/lists/#the-repeat-directive). This directive is usually used for making a keyed list. We can also use it to force re-rendering the auro-icon element when a property changes.

If `this.name` holds the icon name, the following will create a new auro-icon when `this.name` changes, thus re-running the fetch. We accomplish this by making the key the name of the icon, so that Lit understands that a new DOM node should be created. See it live in the [Lit playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCAnQGFsYXNrYWFpcnV4L2F1cm8taWNvbi9kaXN0L2F1cm8taWNvbic7XG5pbXBvcnQge3JlcGVhdH0gZnJvbSAnbGl0L2RpcmVjdGl2ZXMvcmVwZWF0LmpzJztcblxuZXhwb3J0IGNsYXNzIFNpbXBsZUdyZWV0aW5nIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBnZXQgcHJvcGVydGllcygpIHtcbiAgICByZXR1cm4ge1xuICAgICAgbmFtZToge3R5cGU6IFN0cmluZ31cbiAgICB9XG4gIH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICBzdXBlcigpO1xuICAgIHRoaXMubmFtZSA9ICdjcmVkaXQtY2FyZCc7XG4gIH1cbiAgXG4gIGhhbmRsZUNsaWNrKCkge1xuICAgIHRoaXMubmFtZSA9ICdjYy12aXNhJztcbiAgfVxuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgJHtyZXBlYXQoW3RoaXMubmFtZV0sIHggPT4geCwgKG5hbWUpID0-IGh0bWxgXG4gICAgICAgICAgPGF1cm8taWNvbiBjbGFzcz1cImNyZWRpdENhcmQtaWNvblwiIGNhdGVnb3J5PVwicGF5bWVudFwiIG5hbWU9XCIke25hbWV9XCI-PC9hdXJvLWljb24-XG4gICAgICAgIGApfVxuICAgICAgICA8YnV0dG9uIEBjbGljaz0ke3RoaXMuaGFuZGxlQ2xpY2t9PlVwZGF0ZSBpY29uPC9idXR0b24-XG4gICAgYDtcbiAgfVxufVxuXG5jdXN0b21FbGVtZW50cy5kZWZpbmUoJ3NpbXBsZS1ncmVldGluZycsIFNpbXBsZUdyZWV0aW5nKTtcbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9zaW1wbGUtZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8c2ltcGxlLWdyZWV0aW5nPjwvc2ltcGxlLWdyZWV0aW5nPlxuPC9ib2R5PlxuIn1d)

```js
${repeat([this.name], x => x, (name) => html`
  <auro-icon class="creditCard-icon" category="payment" name="${name}"></auro-icon>
`)}
```

IDK if this is more readable than the other option, but it seems to work.

Reply:
As a side note, this trick is basically me implementing the [Svelte key block](https://svelte.dev/docs#key) in Lit.

Reply:
cc @jason-capsule42 and @blackfalcon since I commented after Dale's comment was resolved (left the tab open too long...)